src/main/java/de/qabel/desktop/storage/AbstractNavigation.java

```diff
@@ -248,8 +282,23 @@ private void autocommit() throws QblStorageException {
 		if (!autocommit) {
 			return;
 		}
+		if (autocommitDelay == 0) {
+			commit();	// TODO commitIfModified
+			return;
+		}
 
-		commit();
+		final long autocommitStart = System.currentTimeMillis();
+
+		scheduler.schedule(() -> {
+			try {
+				if (lastAutocommitStart != autocommitStart) {
+					return;
+				}
+				AbstractNavigation.this.commitIfChanged();
```

Comment:
The injected strategy thingy sounds reasonable and could be useful as soon as we need different strategies or at least have to change the strategy. For now it should be fine.